common/config/main.php

```diff
@@ -9,5 +9,13 @@
         'cache' => [
             'class' => 'yii\caching\FileCache',
         ],
```

Comment:
Why not like this?

    'cachePath' => \Yii::getAlias('@common') . '/runtime/cache'

Most code just uses `cache`, which cache to use is very application specific.

Reply:
for BC reason (`Yii::$app->cache`)
I tell about `common\config\main.php`

Also we can duplicate it component like this
```php
        'cache' => [
            'class' => 'yii\caching\FileCache',
        ],
        'cacheConsole' => [
            'class' => \yii\caching\FileCache::class,
            'cachePath' => \Yii::getAlias('@console') . '/runtime/cache'
        ],
        'cacheFrontend' => [
            'class' => \yii\caching\FileCache::class,
            'cachePath' => \Yii::getAlias('@frontend') . '/runtime/cache'
        ],
        'cacheBackend' => [
            'class' => \yii\caching\FileCache::class,
            'cachePath' => \Yii::getAlias('@backend') . '/runtime/cache'
        ],
```
how about this?

or add this configuration to `console\config\main.php` **only** and set as your wish
```php
        'cache' => [
            'class' => 'yii\caching\FileCache',
            'cachePath' => \Yii::getAlias('@console') . '/runtime/cache'
        ],
        'cacheFrontend' => [
            'class' => \yii\caching\FileCache::class,
            'cachePath' => \Yii::getAlias('@frontend') . '/runtime/cache'
        ],
        'cacheBackend' => [
            'class' => \yii\caching\FileCache::class,
            'cachePath' => \Yii::getAlias('@backend') . '/runtime/cache'
        ],
```

Reply:
In both case we need add comment of usage case (short instruction) into `common\config\main.php:9`

Reply:
I see ... while I'd prefer a common cache directory for all apps and just one component configuration by default.

I'd also say that there should be a possibility to flush all caches, if you have different cache folders for the apps, like the current default. 
What's a bit quirky to me is that you have actually two identically configured cache components when you are in a frontend or backend (web-)app.

Reply:
Common cache directory could be a very bad idea depending on how the application is designed. In many cases it may result in unexpected collisions.

Reply:
@samdark Just out of curiosity, could you give me an example for a collision you are referring to?

Reply:
It's writing the same keys from web and CLI app non intentionally.

Reply:
We're using that to populate caches for the web-application via CLI commands. :)
But yeah, it depends on what you expect.